### DISCUSSION TITLE
Qt 6

### DISCUSSION THREAD

**Discussion by BourgeoisLab** [kleiner-brauhelfer/kleiner-brauhelfer-2#296] [General]
Ab heute stellt der KBH auf die Qt Version 6 um (Stand heute Qt 6.5.0). Damit profitiert der KBH von zukünftigen Updates und neuen Features von Qt. Vor allem sollte aber das Problem mit Linux & Wayland damit gelöst werden. Um den Code sauber zu halten, wird es keine Rückwärtskompatibilität zu Qt 5 geben.

Die grösste Konsequenz ist, dass es keine 32bit Versionen für Windows & Linux mehr geben wird. Die Umstellung sollte sonst nahtlos sein... hoffe ich.

**Comment by hurra** [comment]
Alle Anwender, welche die gebundelte Version verwenden (AppImage) haben damit kein Problem.
Die Anwender, die jedoch nativ ein QT über die Distribution verwenden stehen wohl erstmal im Regen, da es bei den Distros recht lang dauert, bis neue Major-Versionen von QT in den Repositories auftauchen. Zumal QT 6.5 erst vor wenigen Tagen veröffentlicht wurde.

Ich kann die Argumente bezüglich des zusätzlichen Aufwands für die Pflege der Rückwärtskompatibilität voll verstehen. Vor allem bei einem Projekt, bei welchem aktuell die Hauptlast der Entwicklung auf einer Person liegt. Hier sollte bezüglich der Developer Happyness keine Kompromisse eingegangen werden :)

Daher blicke ich mit einem weinenden, aber auch mit einem lachenden Auge auf diese Entscheidung.

**Comment by Rasinari** [comment]
Man muss auch so fair sein und abwägen, wieviele von den Nutzern Linux oder XP einsetzen. Es melden sich immer mal wieder User, die noch mit Version 1 unterwegs sind. Seit zehn Jahren weiß man, dass 32-Bit ausläuft. Irgendwann ist dann halt auch gut.

Muss man beim Umstellen von 5.15.2 auf QT 6.5 irgendetwas beachten? Welche Komponenten muss man nochmal zwingend installieren?

**Comment by michaelmorisak** [comment]
Ich hab mal meinen Branch rebased. QT 6.5 installiert und bin glücklich das es nach einigen blöden Fehlern meinerseits auch compilieren und laufenlassen können. 32 Bit ist wirklich nicht mehr aktuell. Sowas gibts ja praktisch gar nicht mehr. Vermutlich ist appimage eh die einzige vernünftige Lösung, denn es gibt auch andere libs (denk nur an libssl oder sowas) die man updaten muss weil sie security bugs haben. 
Wir hatten letztens in der Arbeit ein richtig doofes problem mit QT 4 zeuch das wir vor 10 Jahren gemacht hatten und mit einer alten libssl war die wir nicht mehr haben dürfen aus security Gründen. Weil man das nicht mehr so leicht auf einen neuen Qt portieren kann und weil wir das ding ja eigentlich loswerden wollen war der quickfix mit patchen und Qt compilieren usw wirklich kein Spas. Dahin will man einfach nicht und daher muss man quasi mit der Zeit gehen. 
z.B.: ein qt-webkit ist sicher eine Sache die problematisch ist wenn sie nicht updated wird, denn genau in diesem Bereich kann es zu vielen Problemen kommen. 
Danke @Frederic für die Entscheidung und das Durchführen des Updates. 
So alle 10 Jahre kann man dann schonmal in einen neuen PC investieren ;) 
Wobei 32Bit ja schon eher 20 Jahre ist ;)

**Comment by realholgi** [comment]
Habe die macOS Version mit Qt6.5 mal getestet; läuft.

<img width="612" alt="Bildschirmfoto 2023-04-05 um 10 05 18" src="https://user-images.githubusercontent.com/980419/230020037-4c32dcbc-6863-446b-befb-672f26e2e886.png">

**Comment by BourgeoisLab** [comment]
So wie ich es hier sehe, gibt es für Ubuntu ein Packet für Qt 6.4.2: https://launchpad.net/ubuntu/+source/qt6-base. Da sehe ich also grundsätzlich kein Problem. Mir ist aber nicht klar, wie die Qt Versionen rückwärtkompatibel sind. Also ob der mit Qt 6.5 kompillierte KBH mit den Libraries 6.4 oder auch 6.6 funktioniert.

Falls es kein offizielles Linux Packet gibt, kann man Qt mit dem Skript hier (https://github.com/kleiner-brauhelfer/kleiner-brauhelfer-2/blob/master/README.md#qt-installation-unter-linux) installieren oder mit dem offizillen Qt Installer.

Etwas, dass im Moment nicht funktioniert, ist ausgerechnet das Erstellen der AppImage (inklusive Wayland Unterstüzung) :(. Ich hoffe, das klappt aber noch.

@realholgi: Danke! Ich wollte dich soeben fragen, ob du das testen könntest.

**Comment by Rasinari** [comment]
Ich habe Qt auf 6.5 upgedatet und kann nun nicht mehr Kompilieren. CMake streikt mit Fehlercode 2. Was muss ich tun?

[sub-kleiner-brauhelfer-core-make_first-ordered] Error 2
16:47:54: Der Prozess "/usr/bin/make" wurde mit dem Rückgabewert 2 beendet.
Fehler beim Erstellen/Deployment des Projekts kleiner-brauhelfer-2 (Kit: Qt 6.5.0 for macOS)
Bei der Ausführung von Schritt "Make"

**Reply by realholgi** [reply]
Hmm, du hast vorher das Directory build-macos gelöscht? Komplett?
Ich baue zur Zeit auf arm64 mit QT6.4.1 (habe den Versions-Check qmake File verändert), da ich Qt mit homebrew installiert habe und da  bisher nur 6.4.1 verfügbar ist. Es ist mir sogar schon mal gelungen ein lauffähiges ARM Binary zu bauen, habe es aber bisher nur ein einziges Mal geschafft.

**Comment by Rasinari** [comment]
Was soll ich gelöscht haben? Ich habe nur hier bei mir lokal gelöscht, weil mit Qt 6.5 nichts mehr funktioniert.

Mit Qt 6.5 kann ich das Projekt kleiner-brauhelfer nicht mehr importieren. Bin derzeit ganz raus. Welche Version von Qt nutzt man unter Mac?

Wohlgemerkt: Ich bin kein Programmierer - mache hier nur die Texte. Daher bitte mir so erklären, dass es auch ein Nicht-Programmierer verstehen kann. Was also installiere ich und wie richte ich es dann ein, wenn ich mir den Code kleiner-brauhelfer lade?
Ich muss nur sehen, wie die Texte ausschauen, mehr nicht. Andere Computer, Plattformen interessieren mich nicht. Ich arbeite auf Intel-Mac.

Da klappt jetzt nichts mehr…
